### DISCUSSION TITLE
Add option to run user-defined script after a command

### DISCUSSION THREAD

**Discussion by windexvalence** [RogueOneEcho/caesura#100] [Ideas]
Ideally I'd like for there to be an option to execute a user-defined script after a caesura command. This would be extremely helpful with automatically configuring category assignments in qbittorrent after an `upload` command, as well as simplifying the manual verification process, just as a few examples.

Ideally, this could be something as simple as offering a `--post-script` option for each command, which defaults to null. For example, I'm thinking something like the following:

```shell
docker run --rm \
    -v /my/local_host_path/to_some/script.sh:/user-script.sh \
    ghcr.io/rogueoneecho/caesura upload <source-here> --post-script /my/local_host_path/to_some/script.sh
```

**Comment by RogueOneEcho** [comment]
Is it explicitly for the `upload` command you're thinking?

If so can't you just write a bash script that calls `caesura upload` and then performs your custom logic?

---

If you want to call custom hooks from the `batch` command then you might not know the details of any uploaded torrent so the `post-script` hook is going to need to be called for each upload with arguments for:

-[ ] torrent info hash
-[ ] tracker ID/URL

---

If it's explicitly just setting the category in qBit then #90 is probably a simpler solution

**Reply by RogueOneEcho** [reply]
Closing and marking as won't fix as I don't see the value in this and haven't received a response.

Feel free to suggest reasons for it to be re-opened
